service_crategen/src/commands/generate/codegen/rest_json.rs

```diff
@@ -176,34 +179,46 @@ fn generate_endpoint_modification(service: &Service) -> Option<String> {
 
 // IoT defines a lot of empty (and therefore unnecessary) request shapes
 // don't clutter method signatures with them
-fn generate_method_signature(operation: &Operation, shape: &Shape) -> String {
-    if shape.members.is_some() && !shape.members.as_ref().unwrap().is_empty() {
-        format!(
-            "fn {method_name}(&self, input: {input_type})",
-            method_name = operation.name.to_snake_case(),
-            input_type = operation.input_shape()
-        )
-    } else {
-        format!(
+fn generate_method_signature(operation: &Operation, shape: &Option<&Shape>) -> String {
+    match shape {
+        Some(s) => {
+            if s.members.is_some() && !s.members.as_ref().unwrap().is_empty() {
+                format!(
+                    "fn {method_name}(&self, input: {input_type})",
+                    method_name = operation.name.to_snake_case(),
+                    input_type = operation.input_shape()
+                )
+            } else {
+                format!(
+                    "fn {method_name}(&self)",
+                    method_name = operation.name.to_snake_case()
+                )
+            }
+        }
+        None => format!(
             "fn {method_name}(&self)",
             method_name = operation.name.to_snake_case()
-        )
+        ),
     }
 }
 
 // Figure out what, if anything, should be sent as the body of the http request
-fn generate_payload(service: &Service, input_shape: &Shape) -> Option<String> {
-    let declare_payload = match input_shape.payload {
+fn generate_payload(service: &Service, input_shape: &Option<&Shape>) -> Option<String> {
+    if input_shape.is_none() {
+        return None;
+    }
+    let i = input_shape.unwrap();
```

Comment:
You should be able to simplify everything until this line to `let i = input_shape?;`.